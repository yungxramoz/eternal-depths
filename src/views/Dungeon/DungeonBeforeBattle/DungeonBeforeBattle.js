import React from 'react'
import { useDispatch, useSelector } from 'react-redux'
import RpgButton from '../../../components/atoms/RpgButton/RpgButton'
import { battleStart } from '../../../store/game/gameSlice'

const DungeonBeforeBattle = ({ children }) => {
  const dispatch = useDispatch()
  const stage = useSelector((state) => state.game.stage)

  const startBattle = () => {
    dispatch(battleStart())
  }

  return (
    <>
      <h1>Stage {stage}</h1>
      {children}
      <RpgButton onClick={startBattle} text="Start Battle" />
    </>
  )
}

export default DungeonBeforeBattle
