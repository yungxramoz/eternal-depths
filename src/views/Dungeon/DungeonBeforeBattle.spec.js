import { configureStore } from '@reduxjs/toolkit'
import { fireEvent, render, screen } from '@testing-library/react'
import React from 'react'
import { Provider } from 'react-redux'
import gameReducer from '../../store/game/gameSlice'
import DungeonBeforeBattle from './DungeonBeforeBattle'

describe('DungeonBeforeBattle', () => {
  const store = configureStore({
    reducer: {
      game: gameReducer,
    },
  })
  it('renders correctly', () => {
    render(
      <Provider store={store}>
        <DungeonBeforeBattle />
      </Provider>,
    )
    expect(document.body).toMatchSnapshot('DungeonBeforeBattle')
  })

  it('dispatches battleStart action when "Start Battle" button is clicked', () => {
    store.dispatch = jest.fn()
    render(
      <Provider store={store}>
        <DungeonBeforeBattle />
      </Provider>,
    )
    fireEvent.click(screen.getByText('Start Battle'))
    expect(store.dispatch).toHaveBeenCalledTimes(1)
    expect(store.dispatch).toHaveBeenCalledWith({ type: 'game/battleStart' })
  })
})
