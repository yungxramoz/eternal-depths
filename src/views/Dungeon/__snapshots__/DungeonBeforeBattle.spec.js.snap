// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DungeonBeforeBattle renders correctly: DungeonBeforeBattle 1`] = `
<body>
  <div>
    <h1>
      Stage 
      0
    </h1>
    <button
      class="rpgui-button btn"
    >
      Start Battle
    </button>
  </div>
</body>
`;
